### DISCUSSION TITLE
Installation Issues on Ubuntu

### DISCUSSION THREAD

**Discussion by wesch** [open3e/open3e#104] [Q&A] [answered]
Hi all,
first of all, thanks for this excellent piece of software. mIt is exactly what I'm lokking for.

Unfortunatley I have an issue installing on ubuntu server.
After installation of python3 the installation of the git clone fails:

pip install git+https://github.com/open3e/open3e.git 

Collecting git+https://github.com/open3e/open3e.git
  Cloning https://github.com/open3e/open3e.git to /tmp/pip-kzdio0jz-build
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/lib/python3.6/tokenize.py", line 452, in open
        buffer = _builtin_open(filename, 'rb')
    FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-kzdio0jz-build/setup.py'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-kzdio0jz-build/

Looking into the /tmp directory shows there is no setup.py 

Any ideas or advices?

Best regards 
Wolfgang

**Comment by MyHomeMyData** [comment, marked as answer]
@wesch Hi Wolfgang, thanks for your feedback!

Did you do the installation in venv environment? If not, please try this:
```
python3 -m venv ~/open3e
cd ~/open3e
source ~/open3e/bin/activate
pip install git+https://github.com/open3e/open3e.git
```

Best regards
Jürgen

**Reply by SanMiggel** [reply]
In addition to that make sure that pip runs with the right python version. 
Setup.py is old-fashioned way for installing Python packages. We use a more recent version, hence this file is not there. If you have the right Python version and also the right setup tools it should work. 
i.e
Python version >= 3.9
setuptools>=64
Pip version should also be at least 22

**Reply by wesch** [reply]
Thanks for the swift response. I'm a Python newbie. I'll check this tomorrow.

Best regards
Wolfgang

**Comment by wesch** [comment]
Thanks a lot. It is working now. Unfortunately it was difficult to install a python 3.9 version on my ubuntu server (is still on 18.04). But finally it worked. I followed all the steps from Jürgen. The only thing which seems to be cumbersome is that open3e is only installed in ~/open3e/lib/python3.9/site-packages/open3e.

Is that ok? Nevertheless it works depict and open3e including MQTT is working fine.

Best regards 
Wolfgang

**Reply by MyHomeMyData** [reply]
Good to hear, it's working. Installation in the path you mentioned is fine. It's a feature of python's virtual environments. You may have a look [here](https://wiki.ubuntuusers.de/venv/) or [here](https://docs.python.org/3/library/venv.html) for better understanding of venv.

Best regards
Jürgen

**Reply by wesch** [reply]
thanks a lot. I now have a better understanding of venv. I was confused as this is not mentioned in the wiki. 

Best regards
Wolfgang
